app/services/customers/create_service.rb

```diff
@@ -237,10 +237,10 @@ def handle_api_billing_configuration(customer, params, new_customer)
 
       return unless update_provider_customer
 
+      create_or_update_provider_customer(customer, billing)
+
       if customer.provider_customer&.provider_customer_id
```

Comment:
is this if still needed?

Reply:
It is and yes, we should refactor that part in a followup pull request.